<a href="https://colab.research.google.com/github/santiagovillamizarjaimes-collab/segundocorte/blob/main/previo2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

Nombre: Erick Santiago Villamizar Jaimes

In [2]:
import numpy as
np.random.seed(0)
datos = np.random.randint(161, 199, 50)

**Instrucció 1**

1. Implemente una filtración sobre el elemento datos e imprima el elemento con los valores
resultantes.

In [3]:
datos[datos > 170]

array([180, 182, 197, 184, 185, 185, 173, 184, 185, 178, 198, 186, 174,
       181, 177, 176, 179, 196, 185, 190, 180, 180, 175, 193, 193, 192,
       171, 184, 196, 172, 189, 195, 197, 178, 176])

2. Escale cada uno de los valores de datos utilizando una función lambda:

In [45]:
import pandas as pd
x = datos
x = pd.DataFrame(x)
normalizar = lambda x: (x - x.mean()) / x.std()
resultado = normalizar(x)
print(resultado)


           0
0  -1.540169
1  -1.274623
2  -1.274623
3  -0.743530
4   0.141625
5   0.318656
6   1.646387
7   0.495687
8  -1.009076
9   0.584202
10  0.584202
11 -0.477983
12 -1.451653
13  0.495687
14  0.584202
15 -0.035406
16  1.734903
17  0.672717
18 -0.389468
19 -0.832045
20 -0.743530
21  0.230140
22 -0.123922
23 -1.097592
24 -0.212437
25 -1.540169
26  0.053109
27  1.557872
28  0.584202
29  1.026779
30  0.141625
31  0.141625
32 -0.300953
33  1.292326
34 -1.451653
35 -0.743530
36  1.292326
37  1.203810
38 -0.655014
39  0.495687
40  1.557872
41 -0.566499
42  0.938264
43  1.469357
44 -1.540169
45 -1.540169
46  1.646387
47 -1.097592
48 -0.035406
49 -0.212437


4. Convierta el elemento datos en un DataFrame usando pandas. Imprima el elemento

In [6]:
import pandas as pd

df = pd.DataFrame(datos, columns=["Altura"])
df

,Altura
0,161
1,164
2,164
3,170
4,180
5,182
6,197
7,184
8,167
9,185


3. Implemente el método .describe() e interprete los resultados.


In [8]:
df.describe()

,Altura
count,50.000000
mean,178.400000
std,11.297462
min,161.000000
25%,170.000000
50%,178.500000
75%,185.000000
max,198.000000




*   Las alturas reportadas estan entre 161 cm y 198 cm
*   En promedio la altura es de 178.4 cm


*   El 25% de son inferiores o iguales a 170 cm
*   El 50% de las alturas esta por debajo de 178.5 cm y el otro 50% está por arriba de dicho valor


*   El 75% de las alturas estan por debajo de 185cm


**Instrucción 2**

In [12]:
import pandas as pd

df = pd.DataFrame({
    "nombre": ["Ana", "Luis", "Carlos", "Erick", "Daniela"],
    "Edad": [20, 19, 22, 18, 19],
    "lugar de nacimiento": ["Bucaramanga", "El Socorro", "Piedecuesta", "Floridablanca", "Barichara"],
    "Promedio academico": [4.5, 3.8, 4.2, 4.9, 3.9]
})

1. Muestre las primeras filas usando método .head()

In [13]:
df.head()

,nombre,Edad,lugar de nacimiento,Promedio academico
0,Ana,20,Bucaramanga,4.5
1,Luis,19,El Socorro,3.8
2,Carlos,22,Piedecuesta,4.2
3,Erick,18,Floridablanca,4.9
4,Daniela,19,Barichara,3.9


2. Seleccione una columna específica de su elección

In [14]:
df["lugar de nacimiento"]

,lugar de nacimiento
0,Bucaramanga
1,El Socorro
2,Piedecuesta
3,Floridablanca
4,Barichara


3. Seleccione un conjunto de filas dadas condiciones específicas sobre valores de la columna.


In [17]:
df[df ["Promedio academico"] > 4.1]

,nombre,Edad,lugar de nacimiento,Promedio academico
0,Ana,20,Bucaramanga,4.5
2,Carlos,22,Piedecuesta,4.2
3,Erick,18,Floridablanca,4.9


4. Cree una nueva columna que represente una transformación de una variable.

In [21]:
promedio_semestral= df["Promedio academico"].apply(lambda x: x-0.5)
promedio_semestral

,Promedio academico
0,4.0
1,3.3
2,3.7
3,4.4
4,3.4


In [24]:
df = pd.DataFrame({
    "nombre": ["Ana", "Luis", "Carlos", "Erick", "Daniela"],
    "Edad": [20, 19, 22, 18, 19],
    "lugar de nacimiento": ["Bucaramanga", "El Socorro", "Piedecuesta", "Floridablanca", "Barichara"],
    "Promedio academico": [4.5, 3.8, 4.2, 4.9, 3.9],
    "Promedio Calculo I": promedio_semestral
})
df["Promedio Calculo I"]

,Promedio Calculo I
0,4.0
1,3.3
2,3.7
3,4.4
4,3.4


5. Agrupe los datos por una variable categórica usando el método .groupby().

In [27]:
grupo = df.groupby("nombre")["Promedio academico"].mean()
grupo

,Promedio academico
nombre,
Ana,4.5
Carlos,4.2
Daniela,3.9
Erick,4.9
Luis,3.8


6. Exporte el DataFrame final a un archivo CSV llamado datos_procesados.csv

In [29]:
import csv
df = pd.DataFrame({
    "nombre": ["Ana", "Luis", "Carlos", "Erick", "Daniela"],
    "Edad": [20, 19, 22, 18, 19],
    "lugar de nacimiento": ["Bucaramanga", "El Socorro", "Piedecuesta", "Floridablanca", "Barichara"],
    "Promedio academico": [4.5, 3.8, 4.2, 4.9, 3.9],
    "Promedio Calculo I": promedio_semestral
})

with open("datos_procesados.csv", mode="w", newline="") as f:
  writer = csv.writer(f)
  writer.writerows(df)

In [70]:
from google.colab import files
import csv
datos = [
    ["nombre", "Edad", "lugar de nacimiento", "Promedio academico", "Promedio Calculo I"],
    ["Ana", 20, "Bucaramanga", 4.5, 4.0],
    ["Luis", 19, "El Socorro", 3.8, 3,3],
    ["Carlos", 22, "Piedecuesta", 4.2, 3.7],
    ["Erick", 18, "Floridablanca", 4.9, 4.4],
    ["Daniela", 19, "Barichara", 3.9, 3.4]
]

with open("datos_procesados.csv", "w", newline="") as f:
    writer = csv.writer(f)
    writer.writerows(datos)

files.download("datos_procesados.csv")


<IPython.core.display.Javascript object>

<IPython.core.display.Javascript object>

**Parte III**

1. Genere dos muestras aleatorias de tamaño 5W, donde W corresponden al último dígito del
día de cumpleaños:

 **w=5**

In [38]:
PruebaA = np.random.normal(loc=7.2, scale= 0.4, size= 55)

In [39]:
PruebaB = np.random.gamma(shape=306, scale=0.0229, size=55)

2. Construya un DataFrame en pandas que contenga ambas muestras.


In [42]:
df = pd.DataFrame({
    "Prueba A": PruebaA,
    "Prueba B": PruebaB
})
df

,Prueba A,Prueba B
0,7.154880,7.320708
1,7.562938,6.786387
2,7.526108,7.791994
3,7.291639,6.811004
4,6.789528,7.592579
5,7.391010,7.726659
6,7.717079,7.396588
7,6.907417,7.662127
8,6.557839,6.446487
9,7.595790,6.609967


3. Realice un análisis descriptivo para cada una de las pruebas: Media, Mediana, máximo,
mínimo, desviación estándar, cuantil 25% y cuantil 75%.


In [43]:
df.describe()

,Prueba A,Prueba B
count,55.000000,55.000000
mean,7.224556,7.006797
std,0.383209,0.463570
min,6.506016,5.909656
25%,6.975746,6.720296
50%,7.211011,6.986468
75%,7.481777,7.396350
max,8.119909,7.916687


Las notas más baja de los estudiantes de la prueba A fue de 6.5, mientras que een la prueba B la nota más baja fue de 5.9.

Tanto para la prueba A como para la prueba B las calificacion maás alta es un valor bastante cercano, sin embargo, en la prueba A se obtuvi un valor más alto.

En la prueba A el 50% de los estudiantes tuvo notas inferiores a 7.2 mientras que la otra mitad tuvo notas superiores a dicho valor

En la prueba B el 50% de los estudiantes tuvo notas inferiores a 6.9 mientras que la otra mitad tuvo notas superiores a dicho valor

En promedio los estudiantes tuvieron una nota de 7.2, con una desviación estandar de 0.38 pumtos en la prueba A. Para la prueba B el promedio fue de 7, con una desviación esttándar de 0.46 puntos.



4. Aplique prueba de normalidad de Shapiro-Wilk para cada prueba ¿se cumple el supuesto
de normalidad con una confianza del 95%

In [53]:
from scipy.stats import shapiro
shapiro(df["Prueba A"])


ShapiroResult(statistic=np.float64(0.9854101039283362), pvalue=np.float64(0.7408869781355921))

**No se rechaza** la hipótesis de normalidad con una confianza del 95%, por lo que se cumple el supuesto de normalida.

In [54]:
from scipy.stats import shapiro
shapiro(df["Prueba B"])

ShapiroResult(statistic=np.float64(0.9835528506747372), pvalue=np.float64(0.6514030077948881))

**No se rechaza** la hipótesis de normalidad con una confianza del 95%, por lo que se cumle el supuesto de normalidad

5. Aplique la prueba t Student para muestras paradas: use la función stats.ttest_rel
¿Se rechaza que el puntaje medio en ambas encuestas es igual con una confianza del 95%?

In [58]:
from scipy import stats
stats.ttest_rel(df["Prueba A"], df["Prueba B"])

TtestResult(statistic=np.float64(2.596453256064248), pvalue=np.float64(0.01210913971186413), df=np.int64(54))

**Se rechaza** que el puntaje medio en ambas encuestas es igual con una confianza del 95%

6. ¿Se rechaza con una confianza del 95% que los puntajes de ambas encuestas no están
correlacionados?

In [59]:
stats.kendalltau(df["Prueba A"], df["Prueba B"])

SignificanceResult(statistic=np.float64(-0.05454545454545456), pvalue=np.float64(0.5565170247628305))

**No se rechaza** con una confianza del 95% que los puntajes de ambas encuestas no están correlacionados

7. Aplique la prueba con el coeficiente de correlación de Spearman: función.spearmanr
¿Se rechaza con una confianza del 95% que los puntajes de ambas encuestas no están correlacionados?


In [62]:
stats.spearmanr(df['Prueba A'], df['Prueba B'])

SignificanceResult(statistic=np.float64(-0.07626262626262627), pvalue=np.float64(0.5799929383317582))

**No se rechaza** con una confianza del 95% que los puntajes de ambas encuestas no están correlacionados. Es decir, que se acepta con una confianza del 95% que los puntajes de ambas encuestas están correlacionados

8. Utilizando los códigos anteriores cree un paquete en Google Colab llamado comparacion:


In [65]:
import os
os.makedirs("comparacion", exist_ok=True)

9. Cree dos módulos:

In [66]:
%%writefile comparacion/pruebat.py

def prueba_t():
  from scipy import stats
  return stats.ttest_rel(df["Prueba A"], df["Prueba B"])

Writing comparacion/pruebat.py


In [67]:
%%writefile comparacion/cor.py

def coeficiente_K():
  return stats.kendalltau(df["Prueba A"], df["Prueba B"])

def coeficiente_s():
  return stats.spearmanr(df['Prueba A'], df['Prueba B'])

Writing comparacion/cor.py


10. Utilice el paquete construido en el ítem 8 y los módulos del ítem 9 para analizar el
siguiente conjunto de datos, el cual ya se encuentra organizado como un DataFrame:

In [68]:
df = pd.DataFrame({
 "Estudiante": range(1, 11),
 "Prueba_A": [7.1, 7.5, 6.8, 7.3, 7.0, 7.4, 6.9, 7.2, 7.6, 7.1],
 "Prueba_B": [6.9, 7.2, 7.0, 7.1, 6.8, 7.3, 7.0, 7.1, 7.4, 7.0]
})